account/manager.go

```diff
@@ -2080,3 +2226,24 @@ func makeTxnLabel(labelPrefix, contextLabel string) string {
 
 	return label
 }
+
+// serializeWitness turns a wire witness into its serialized form.
+func serializeWitness(witness wire.TxWitness) ([]byte, error) {
+	var buf bytes.Buffer
+	if err := psbt.WriteTxWitness(&buf, witness); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+// sigHashForScript returns the sighash flag for the given UTXO's pkScript.
```

Comment:
May be worth [adding this to btcd](https://github.com/btcsuite/btcd/pull/1868) for default script types